gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -1492,6 +1510,11 @@ private String listStorageObjectsAndPrefixesPage(
         logger.atFiner().withCause(e).log(
             "listStorageObjectsAndPrefixesPage(%s, %s): item not found", resource, listOptions);
         return null;
+      } else if (ApiErrorExtractor.INSTANCE.requestFailure(e)) {
+        gcsStatisticsMap.put(
+            GoogleCloudStorageStatistics.ACTION_HTTP_GET_REQUEST_FAILURES,
+            gcsStatisticsMap.get(GoogleCloudStorageStatistics.ACTION_HTTP_GET_REQUEST_FAILURES)
+                + 1L);
```

Comment:
Avoid intrusive changes, let's introduce a different class to handle this logic as done in PR 613